src/main/java/com/perimeterx/models/PXContext.java

```
@@ -206,10 +207,10 @@ public class PXContext {
     private String pxhd;
     private String responsePxhd;
     private boolean isMonitoredRequest;
-    private AdditionalContext additionalContext;
+    private LoginData loginData;
     private UUID requestId;
 
-    public PXContext(final HttpServletRequest request, final IPProvider ipProvider, final HostnameProvider hostnameProvider, PXConfiguration pxConfiguration) {
+    public PXContext(final HttpServletRequest request, final IPProvider ipProvider, final HostnameProvider hostnameProvider, PXConfiguration pxConfiguration) throws PXException {
```

This new exception concerns me -- a failed credentials extraction shouldn't cause an exception in the PXContext. It should fail silently.

---

Answered above.